orchagent/dash/dashorch.cpp

```diff
@@ -202,16 +196,13 @@ bool DashOrch::addApplianceEntry(const string& appliance_id, const dash::applian
     }
     direction_lookup_attrs.push_back(appliance_attr);
 
-    appliance_attr.id = SAI_DIRECTION_LOOKUP_ENTRY_ATTR_DASH_ENI_MAC_OVERRIDE_TYPE;
-    if (entry.has_outbound_direction_lookup())
-    {
-        appliance_attr.value.u32 = sMacOverride[entry.outbound_direction_lookup()];
-    }
-    else
+    // Don't set up this entry for dst_mac i.e. for Floating NIC.
+    if (!entry.has_outbound_direction_lookup() || (entry.outbound_direction_lookup() != "dst_mac"))
     {
+        appliance_attr.id = SAI_DIRECTION_LOOKUP_ENTRY_ATTR_DASH_ENI_MAC_OVERRIDE_TYPE;
```

Comment:
the current design is that:
1. If outbound direction lookup is not set, we default to SRC MAC, no override needs to be provided.
2. If outbound direction lookup is set, we set to whatever the value is.

Here is the HLD link, just in case: https://github.com/sonic-net/DASH/blob/main/documentation/express-route-service/express-route-gateway-bypass.md#51-eni-mac-handling